### PR TITLE
fix leader nullpointexception and increament listener core pool size

### LINKED ISSUE #2122: Frequently creation and shutdown cause zookeeper watch to be lost.
### Which version of ElasticJob did you use?
master

### Which project did you use? ElasticJob-Lite or ElasticJob-Cloud?
ElasticJob-Lite

### Expected behavior
every event in zookeeper, client can watch it, otherwise exception occur

### Actual behavior
frequently create and shutdown Immediately，watch lost

### Reason analyze (If you can)
project has 12 listener now, every one can watch event, but single thread pool core size is not enough, when job shutdown, shudownListener maybe execute over first, but GuaranteeListener maybe not get complete signal，lead to AbstractDistributeOnceElasticJobListener appear timeout exception，so should be increament core size to avoid watch loss.

### Steps to reproduce the behavior.
extends a AbstractDistributeOnceElasticJobListener,  set complete or start timeout, Frequently creation and shutdown some job

### Example codes for reproduce this issue (such as a github link).

### PR DESCRIPTION
**Pull request by xlcn**
Fixes #2122 . 

Changes proposed in this pull request:
-
-
-
when job shutdown,  code should check before elect leader.
increament listener threadpool core size , avoid watch lost

### CODE REVIEW

**Comment by TeslaCN** [comment]
How about using unbounded thread pool?

**Comment by xlcn** [comment]
Thank you for your advice. please recheck.

File: elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/listener/ListenerNotifierManager.java [hunk 1]
```diff
@@ -63,7 +63,8 @@ public void registerJobNotifyExecutor(final String jobName) {
             synchronized (this) {
                 if (!listenerNotifyExecutors.containsKey(jobName)) {
                     ThreadFactory threadFactory = ThreadUtils.newGenericThreadFactory("ListenerNotify-" + jobName);
-                    ExecutorService notifyExecutor = Executors.newSingleThreadExecutor(threadFactory);
+                    // Use unbounded thread pool to cover all the listeners to execute event, avoid zk watch loss.
```
Comment (TeslaCN) [inline elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/listener/ListenerNotifierManager.java]: The comment could be removed.

**Comment by TeslaCN** [comment]
Please fix the CI.

File: elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/listener/ListenerNotifierManager.java (hunk 1 above)
Comment (xlcn) [reply elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/listener/ListenerNotifierManager.java]: okay

**Comment by xlcn** [comment]
retry? local machine (mac) test is okay

**Comment by TeslaCN** [comment]
> retry? local machine (mac) test is okay

That means the unit tests are not stable.

**Comment by xlcn** [comment]
> > retry? local machine (mac) test is okay
> 
> That means the unit tests are not stable.

timeout again _^!

**Comment by xlcn** [comment]
@TeslaCN could you rerun CI action? Thx

**Pull request closed without merging**